### PR TITLE
[WIP] feat: upgrade react native to 64.2

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **feat**

### Description

This PR upgrades react-native to 64.2. My goal is to get the react-native higher as much as I can to make it easier for me to update to reanimated 2 which will require turbo modules to be there running.
I decided to upgrade version by version to avoid breaking many stuff at once. Nonethelss, I faces issues literally at every possible step on the way, then it got personal 😅

BTW, I realized we have a lot of tests that are breaking locally because we didn't set the timezone properly and the formatting was different. It would be great to fix those

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Upgrade react-native to version 0.64.2 with react 17 support - mounir

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 428e5093da69ced3b26cfd226330ea504ce761b3
</p>

File: patches/@wojtekmaj+enzyme-adapter-react-17+0.6.6.patch [hunk 1]
```diff
@@ -0,0 +1,9 @@
+diff --git a/node_modules/@wojtekmaj/enzyme-adapter-react-17/index.d.ts b/node_modules/@wojtekmaj/enzyme-adapter-react-17/index.d.ts
+index a991d3e..aa7dc75 100644
+--- a/node_modules/@wojtekmaj/enzyme-adapter-react-17/index.d.ts
++++ b/node_modules/@wojtekmaj/enzyme-adapter-react-17/index.d.ts
+@@ -1,3 +1,4 @@
++// @ts-ignore
+ import { EnzymeAdapter } from 'enzyme';
```
Comment (MounirDhahri) [inline patches/@wojtekmaj+enzyme-adapter-react-17+0.6.6.patch:7]: we are ignoring this everywhere in Eigen and since it wasn't ignored here it was breaking tsc in ci

File: package.json [hunk 2]
```diff
@@ -269,19 +269,19 @@
...
+    "metro-react-native-babel-preset": "0.64.0",
     "node-fetch": "2.6.7",
     "ora": "5.4.0",
     "patch-package": "6.4.7",
     "postinstall-prepare": "1.0.1",
     "prettier": "2.5.1",
     "pull-lock": "1.0.0",
     "puppeteer": "7.1.0",
-    "react-dom": "16.8.3",
+    "react-dom": "17.0.1",
```
Comment (MounirDhahri) [inline package.json:280]: this was the reason why I failed many times before at the upgrade I belive 😄

File: package.json [hunk 3]
```diff
@@ -145,10 +145,10 @@
     "ordinal": "1.0.3",
     "prop-types": "15.7.2",
     "query-string": "4.3.4",
-    "react": "16.13.1",
+    "react": "17.0.1",
```
Comment (MounirDhahri) [inline package.json:148]: what's uuup

File: emission/Emission.podspec [hunk 4]
```diff
@@ -56,7 +56,7 @@ podspec = Pod::Spec.new do |s|
   # s.dependency 'Yoga', "#{react_native_version}.React"
   react_podspecs = [
     '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec',
-    '../node_modules/react-native/third-party-podspecs/Folly.podspec',
+    '../node_modules/react-native/third-party-podspecs/RCT-Folly.podspec',
```
Comment (MounirDhahri) [inline emission/Emission.podspec:59]: react-native updated folly spec name and i had to update it here as well since are explicitly importing it for Emission

File: src/app/Scenes/Gene/Gene.tests.tsx [hunk 5]
```diff
@@ -66,13 +63,4 @@ describe("Gene", () => {
...
-
-  it("renders filter modal", () => {
-    const tree = renderWithWrappers(<TestRenderer />)
-    mockEnvironmentPayload(environment)
-
-    act(() => tree.root.findByType(TouchableHighlightColor).props.onPress())
-
-    expect(tree.root.findAllByType(ArtworkFilterOptionsScreen)).toHaveLength(1)
-  })
 })
```
Comment (MounirDhahri) [inline src/app/Scenes/Gene/Gene.tests.tsx:66]: I could have solved this by simply adding `await flushPromiseQueue()` but the test was duplicate and not needed

Comment (MounirDhahri) [inline src/app/Scenes/Gene/Gene.tests.tsx:66]: I could have solved this by simply adding `await flushPromiseQueue()` but the test was duplicate and not needed

File: src/setupJest.ts [hunk 6]
```diff
@@ -165,14 +165,18 @@ const originalConsoleError = console.error
...
+        /Warning: React does not recognize the `\w+` prop on a DOM element\./.test(message) ||
+        /Warning: The tag <\w+> is unrecognized in this browser\./.test(message) ||
+        /Warning: Unknown event handler property `\w+`\./.test(message) ||
+        /Warning: An update to [\w\s]+ inside a test was not wrapped in act/.test(message) ||
+        /Warning: Received `\w+` for a non-boolean attribute `\w+`\./.test(message) ||
+        /Warning: [\w\s]+ has been extracted from react-native core/.test(message))) ||
+    /use `useFragment` instead of a Relay Container/.test(message) ||
+    /Warning: RelayResponseNormalizer:/.test(message) ||
+    /Warning: RelayModernRecord:/.test(message) ||
+    /Use PascalCase for React components/.test(message)
```
Comment (MounirDhahri) [inline src/setupJest.ts:179]: These were making the list of warning crazy in Jest and we just swallow them here to make the list humanly readable 😄  Of course these are only going to not shown when the test passes successfully

File: package.json [hunk 7]
```diff
@@ -255,12 +256,11 @@
     "dedent": "0.7.0",
     "dotenv": "8.2.0",
     "enzyme": "3.9.0",
-    "enzyme-adapter-react-16": "1.13.0",
```
Comment (MounirDhahri) [inline package.json:258]: bye 👋

File: package.json [hunk 8]
```diff
@@ -239,10 +239,11 @@
     "@types/styled-system__theme-get": "5.0.2",
     "@types/supercluster": "5.0.3",
     "@types/yup": "0.29.13",
+    "@wojtekmaj/enzyme-adapter-react-17": "0.6.6",
```
Comment (MounirDhahri) [inline package.json:242]: See this issue https://github.com/enzymejs/enzyme/issues/2429
